### PR TITLE
feat: add remove method and deprecate clear

### PR DESCRIPTION
**Pull request by boschni**

### CODE REVIEW

File: src/core/query.ts [hunk 1]
```diff
@@ -206,7 +206,14 @@ export class Query<TResult, TError> {
     })
   }
 
+  /**
+   * @deprecated
+   */
   clear(): void {
```
Comment (tannerlinsley) [inline src/core/query.ts:212]: Can we add a console log here to warn?

Comment (boschni) [reply src/core/query.ts:212]: Added

**Comment by renatobenks** [comment]
shouldn't it be a major release instead? 🤔 it's a breaking change, isn't it?

**Comment by tannerlinsley** [comment]
The functionality is exactly the same for now. The only difference is you will see a deprecation notice until you change to the new function name.

**Pull request merged**